scripts/run-mod-settings.sh

```diff
@@ -0,0 +1,184 @@
+curl -d'{"name":"DB_USERNAME","value":"postgres"}' $U/_/env
+curl -d'{"name":"DB_PASSWORD","value":"postgres3636"}' $U/_/env
+curl -d'{"name":"DB_DATABASE","value":"postgres"}' $U/_/env
+
+# Set of modules that are necessary to bootstrap admin user
+CORE_MODULES="mod-users mod-login mod-permissions"
+
+TEST_MODULES="mod-settings"
+
+compile_module() {
+	local m=$1
+	if test ! -d $m; then	
+		git clone --recurse-submodules git@github.com:folio-org/$m
+
+	fi
+	if test ! -d $m; then
+		echo "$m missing. git clone failed?"
+		exit 1
+	fi
+	cd $m
+	mvn -DskipTests -Dmaven.test.skip=true verify
+	cd ..
+}
+register_module() {
+	local m=$2
+	echo "Register module $m"
+	local md=$m/target/ModuleDescriptor.json
+	if test ! -f $md; then
+		compile_module $m
+	fi
+	if test ! -f $md; then
+		echo "$md missing pwd=`pwd`"
+		exit 1
+	fi
+	if test "$1" != "x"; then
+		OPT=-HX-Okapi-Token:$1
+	else
+		OPT=""
+	fi
+	curl -s $OPT -d@$md $U/_/proxy/modules -o /dev/null
+	local dd=$m/target/DeploymentDescriptor.json
+}
+
+deploy_module() {
+	local m=$2
+	echo "Deploy module $m"
+	if test "$1" != "x"; then
+		OPT=-HX-Okapi-Token:$1
+	else
+		OPT=""
+	fi
+	local dd=$m/target/DeploymentDescriptor.json
+	curl -s $OPT -d@$dd $U/_/deployment/modules -o /dev/null
+}
+
+deploy_modules() {
+	for m in $2; do
+		register_module $1 $m
+	done
+	for m in $2; do
+		deploy_module $1 $m
+	done
+}
+
+install_modules() {
+	local j="["
+	local sep=""
+	for m in $3; do
+		j="$j $sep {\"action\":\"$2\",\"id\":\"$m\"}"
+		sep=","
+	done
+	j="$j]"
+	if test "$1" != "x"; then
+		OPT=-HX-Okapi-Token:$1
+	else
+		OPT=""
+	fi
+	curl -s $OPT "-d$j" "$U/_/proxy/tenants/$T/install?purge=true&tenantParameters=loadReference%3Dtrue%2CloadSample%3Dtrue"
+}
+
+okapi_curl() {
+	if test "$1" != "x"; then
+		local OPT="-HX-Okapi-Token:$1"
+	else
+		local OPT="-HX-Okapi-Tenant:$T"
+	fi
+	shift
+	curl -s $OPT -HContent-Type:application/json $*
+}
+
+make_adminuser() {
+	local username=$2
+	local password=$3
+	
+	uid=`uuidgen`
+	okapi_curl $1 -XDELETE "$U/users?query=username%3D%3D$username"
+	okapi_curl $1 -d"{\"username\":\"$username\",\"id\":\"$uid\",\"active\":true}" $U/users
+	okapi_curl $1 -d"{\"username\":\"$username\",\"userId\":\"$uid\",\"password\":\"$password\"}" $U/authn/credentials
+	puid=`uuidgen`
+	okapi_curl $1 -d"{\"id\":\"$puid\",\"userId\":\"$uid\",\"permissions\":[\"okapi.all\",\"perms.all\",\"users.all\",\"login.item.post\",\"perms.users.assign.immutable\"]}" $U/perms/users
+}
+
+login_admin() {
+	curl -s -Dheaders -HX-Okapi-Tenant:$T -HContent-Type:application/json -d"{\"username\":\"$username\",\"password\":\"$password\"}" $U/authn/login
+token=`awk '/x-okapi-token/ {print $2}' <headers|tr -d '[:space:]'`
+}
+
+deploy_modules x "$CORE_MODULES"
+
+deploy_modules x mod-authtoken
+
+install_modules x enable "$CORE_MODULES"
+install_modules x enable okapi
+
+make_adminuser x $username $password
+
+install_modules x enable mod-authtoken
+
+login_admin
+
```

Comment:
OK, I am intrigued. This is not a PR review comment, but I would like to understand why we need a valid session token to deploy and install the test-modules, but did not need one to deploy and install the core modules and mod-authtoken.

Reply:
Because Mod-authtoken is enabled

Reply:
D'oh! Of course. It's nothing to do with which modules are being deployed/installed, it's to do when when in the sequence that's happening.